daemon/list.go

```diff
@@ -133,7 +133,18 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			ID:    container.ID,
 			Names: names[container.ID],
 		}
-		newC.Image = container.Config.Image
+
+		img, err := daemon.Repositories().LookupImage(container.Config.Image)
+		if err != nil {
+			// If the image can no longer be found by its original reference,
+			// it makes sense to show the ID instead of a stale reference.
+			newC.Image = container.ImageID
```

Comment:
@relistan I'm still new to this, so please don't interpret this as me being snarky. Here are the two edge cases I ran up against when trying to get this to work, could we make these work without doing `LookupImage`?

1. Intermediate build containers where `container.Config.Image == ""`. This was causing me problems in `TestBuildContextCleanupFailedBuild`.
1. Situations in which a container is created using repo:tag and then that tag is deleted. Failing test for me here was `TestRmContainerOrphaning`.

Happy to be educated. :)